ARCs/arc-0014.md

```diff
@@ -0,0 +1,246 @@
+
+#### Transaction Authentication Message
+
+A *Transaction Authentication Message* is an *Authentication Message* represented as an Algorand `Transaction` object. Such a transaction must have the following characteristics:
+
+- it **MUST** be an Algorand `Payment Transaction`;
+- it **MUST** include a *Simple Authentication Message* into the transaction's `note` field;
+- it **MUST** be *invalidated* i.e. not executable on any official Algorand network.
+
+> Malicious users could intercept and execute it causing unexpected consequences. For instance, in case of a **MainNet** transaction, it could be intercepted by a malicious user and executed, burning some Algos from the sender’s account due to the payment of fees.
+
+The fields of a `Payment Transaction` object which represents a *Transaction Authentication Message* **MUST** be initialized as follows
+
+```
+amount = 0;
+sender/receiver = *PKa*;
+firstValid/lastValid = 1;
+fee = 0;
+genesisId = “arc14-arc”;
+genesisHash = SHA-512/256 of the string “arc14-auth”;
+note = “arc14”+msgpacked_auth_message;
+```
+The fields `genesisId` and `genesisHash` specify respectively the id and hash of the genesis block of the network, and they **MUST** be initialized to the ARC-14 values (not conventional for MainNet/TestNet/BetaNet). The `note` field **MUST** include a *Simple Authentication Message* object. Finally, the `sender` field **MUST** be set to the Algorand address of the user.
+
+> A detailed description of the transitions fields of Algorand Transactions is available in the transaction's reference <a href="https://developer.algorand.org/docs/get-details/transactions/transactions/">documentation</a>.
+
+For example:
+
+```json
+{
+  "txn": {
+    "amt": 0,
+    "fee": 0,
+    "fv": 1,
+    "gen": "arc14-auth",
+    "gh": "f20b122eb226626b5c0d36cd33edcdb94613820f048baf2a1d6dfe46e5be18d1",
+    "lv": 1,
+    "rcv": "EW64GC6F24M7NDSC5R3ES4YUVE3ZXXNMARJHDCCCLIHZU6TBEOC7XRSBG4",
+    "snd": "EW64GC6F24M7NDSC5R3ES4YUVE3ZXXNMARJHDCCCLIHZU6TBEOC7XRSBG4",
+    "type": "pay",
+    "note": "arc14<msgpacked_auth_message>",
+  }
+}
+```
+
+### How to sign an authentication message?
+
+#### Digital Signature of a Simple Authentication Message
+
+A *Simple Authentication Message* can be signed with any open-source cryptographic library (e.g. Python lib PyNaCl) and the secret key of an Algorand account. Algorand account’s secret key *SKa* represents a base64 concatenation of the traditional 32-bytes secret and public keys. To create a digital signature of a *Simple Authentication Message* the Algorand secret key **MUST** be decoded as a standard 32-bytes secret key as follows:
+
+1. base64 decode the Algorand secret key *SKa*;
+2. extract the traditional secret key *SK* as the first 32 bytes of the decoded key.
+
+#### Digital Signature of a Transaction Authentication Message
```

Comment:
Wouldn't it be more secure to support only "Simple Authentication Message"?

I think supporting two different versions of authentication can be difficult for dApp developers and users. I am unable to clearly visualize the user flow in the scenario that we allow 2 type of authentication.

I assume on dApps, there will be 2 different buttons: "Authenticate with Simple Authentication Message" and "Authenticate with Transaction Authentication Message". For the end user, this can be quite confusing and it can open  door to various attacks. User may not know if their wallet supports Simple Authentication Message or not.

I believe that:
 - Announcing the format of Simple Authentication to wallets, with the help of a demo app
 - Giving wallets a couple of months of implementation window.
 - Announcing ARC-14 standard to dApp devs & releasing it for the end users
can give the best result in the long run.

Do you have any idea on the possible UX alternatives with 2 different methods?

Reply:
Yigit - what are your thoughts on CTAP2 / WebAuthN and FIDO2 functionality overlap with these sorts of proposals?

Reply:
@yigitguler I agree that a demo app could a lot. I am thinking whether wallets already offer the possibility of signing random messages. Walletconnect and MyAlgo seem ready:
- https://connect.myalgo.com/docs/getting-started/myalgo-connect-api#signbytes 
- https://docs.walletconnect.com/client-api#sign-message-eth_sign

Ledger is not fully supported though.